code/iaas/auth-logic/src/main/java/io/cattle/platform/iaas/api/auth/projects/ProjectMemberResourceManager.java

```diff
@@ -55,14 +55,23 @@ protected Object listInternal(SchemaFactory schemaFactory, String type, Map<Obje
         Policy policy = (Policy) ApiContext.getContext().getPolicy();
         String id = RequestUtils.makeSingularStringIfCan(criteria.get("id"));
         if (StringUtils.isNotEmpty(id)) {
-            ProjectMember projectMember = authDao.getProjectMember(Long.valueOf(id));
+            ProjectMember projectMember;
+            try {
+                 projectMember = authDao.getProjectMember(Long.valueOf(id));
+            } catch (NumberFormatException e) {
+                return null;
```

Comment:
Raising the exception ends the request. If it is a request that is using the manager internally it would end that request which is bad. The 404 that is raised here is auth related but if they try to list something that doesn't exist they shouldn't lose their request.
Since other places internally use the projectManager to list members.